clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -284,7 +283,8 @@ public KafkaProducer(final Map<String, Object> configs) {
      *                         be called in the producer when the serializer is passed in directly.
      */
     public KafkaProducer(Map<String, Object> configs, Serializer<K> keySerializer, Serializer<V> valueSerializer) {
-        this(configs, keySerializer, valueSerializer, null, null, null, Time.SYSTEM);
+        this(new ProducerConfig(ProducerConfig.addSerializerToConfig(configs, keySerializer, valueSerializer)),
```

Comment:
The previous approach was intended to avoid having this logic in many places.

Reply:
Thanks for reviews!

Do you mean that previous approach tried to avoid ‘’’addSerializerToConfig’’’? If so, should we do it for KafkaConsumer?

Reply:
Yeah, it's a bit error prone to have that logic in every constructor. We could move the `propsToMap` method to a utility class and use it on the consumer too.